design/cstor/v1/cstorpool.md

```diff
@@ -0,0 +1,504 @@
+# Moving CSPC and CSPI Schema to cstor group and version v1 from openebs group and version v1alpha1
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [Current State of Things](#current-state-of-things)
+* [CSPC and CSPI Schema Proposal](#cspc-schema-proposal)
+* [CRD Migration](#crd-migration)
+
+## Introduction
+
+This proposal highlights the limitations of current CSPC and CSPI schema and
+proposes improvements for the same.
+Please refer to the following link for cstor operator design document. 
+https://github.com/openebs/openebs/blob/master/contribute/design/1.x/cstor-operator/doc.md
+
+# Goals
+
+The major goal of this document is to freeze the schema for CSPC and CSPI schema.
+Apart from this the document focuses on following aspects too:
+
+- Moving the CSPC and CSPI CRD to v1 api version and cstor api group.
+- Enhance the API in order to enhance user experience and support new features.
+
+
+## Current State of Things
```

Comment:
This whole section can be removed. No need for previous version related limitations in this PR.